src/pudl/output/eia923.py

```diff
@@ -156,9 +159,79 @@ def generation_fuel_eia923(pudl_engine, freq=None,
     return out_df
 
 
-def fuel_receipts_costs_eia923(pudl_engine, freq=None,
-                               start_date=None, end_date=None,
-                               fill=False, roll=False):
+def generation_fuel_all_eia923(gf: pd.DataFrame, gfn: pd.DataFrame) -> pd.DataFrame:
+    """
+    Combine nuclear and non-nuclear generation fuel tables into a single output.
+
+    The nuclear and non-nuclear generation fuel data are reported at different
+    granularities. For non-nuclear generation, each row is a unique combination of date,
+    plant ID, prime mover, and fuel type. Nuclear generation is additionally split out
+    by nuclear_unit_id (which happens to be the same as generator_id).
+
+    This function aggregates the nuclear data across all nuclear units within a plant so
+    that it is structurally the same as the non-nuclear data and can be treated
+    identically in subsequent analyses. Then the nuclear and non-nuclear data are
+    concatenated into a single dataframe and returned.
+
+    Args:
+        gf: non-nuclear generation fuel dataframe.
+        gfn: nuclear generation fuel dataframe.
+
+    """
+    primary_key = [
+        "report_date",
+        "plant_id_eia",
+        "prime_mover_code",
+        "energy_source_code",
+    ]
+    sum_cols = [
+        "fuel_consumed_for_electricity_mmbtu",
+        "fuel_consumed_for_electricity_units",
+        "fuel_consumed_mmbtu",
+        "fuel_consumed_units",
+        "net_generation_mwh",
+    ]
+    replace_cols = ["nuclear_unit_id", "fuel_mmbtu_per_unit"]
```

Comment:
Hmm, neither of these feels right to me since we're really doing different things to these two columns. One is recalculated and the other is dropped through the groupby. Maybe `other_cols` and a comment.

Reply:
i hear you, but in this context you are effectively removing them.